src/lexer.rs

```diff
@@ -239,30 +244,29 @@ impl<'a> Lexer<'a> {
         self.had_comments
     }
 
-    #[allow(clippy::unnecessary_wraps)]
     fn parse_identifier(&mut self) -> Result<TokenValue, PreprocessorError> {
         let mut identifier = String::default();
 
-        while let Some(&(current, _)) = self.inner.peek() {
-            match current {
-                'a'..='z' | 'A'..='Z' | '_' | '0'..='9' => {
-                    self.inner.next();
-                    identifier.push(current);
-                }
-                _ => {
-                    break;
-                }
-            }
+        if self
```

Comment:
nit: This is technically out of spec for GLSL. Can we have a TODO to add a runtime option for whether to support unicode identifiers or not?

Reply:
I asked on the Naga channel if WGSL will support unicode identifiers. If it does I think we should keep it since eventually some folks using WGSL might want to use `pp-rs` to preprocess shaders outside of Naga,